src/common/libflux/disconnect.h

```diff
@@ -17,11 +17,23 @@
 extern "C" {
 #endif
 
+/* Convenience function for ensuring messages match for disconnect.
+ * Effectively calls flux_msg_match_route_first() and
+ * flux_msg_cred_authorize().
+ */
```

Comment:
Might be clearer to say "Return true if disconnect request `msg1` came from same sender as `msg2' and has appropraite authorization".